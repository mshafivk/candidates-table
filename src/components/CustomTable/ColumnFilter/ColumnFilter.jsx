import React from 'react';
import PropTypes from 'prop-types';
import { useFilterEntries } from '../../../hooks/useFilterEntries';
import classes from './ColumnFilter.less';
function ColumnFilter({ data, selector, onFilterChange }) {
  const filterEntries = useFilterEntries(data, selector);
  return (
    <select
      className={classes.dropdown}
      data-testid="filter-dropdown"
      onChange={(e) => onFilterChange(selector, e.target.value)}>
      {filterEntries.map((item, index) => (
        <option value={item} key={index}>
          {item}
        </option>
      ))}
    </select>
  );
}

export default React.memo(ColumnFilter);

ColumnFilter.propTypes = {
  data: PropTypes.array,
  selector: PropTypes.string,
  onFilterChange: PropTypes.func,
};
