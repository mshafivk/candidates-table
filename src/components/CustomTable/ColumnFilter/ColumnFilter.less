.dropdown {
  padding: 4px 0px;
  margin-top: 6px;
}
