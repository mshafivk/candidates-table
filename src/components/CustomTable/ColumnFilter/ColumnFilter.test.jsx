import React from 'react';
import ColumnFilter from './ColumnFilter';
import { shallow } from 'enzyme';

describe('ColumnFilter Component', () => {
  const filterChangeHandler = jest.fn();
  const props = {
    data: ['All', 'awaiting', 'rejected'],
    selector: 'status',
    onFilterChange: filterChangeHandler,
  };
  it('should render properly', () => {
    const wrapper = shallow(<ColumnFilter {...props} />);

    wrapper
      .find('[data-testid="filter-dropdown"]')
      .at(0)
      .simulate('change', { target: { value: 'awaiting', name: 'awaiting' } });
    expect(filterChangeHandler).toHaveBeenCalled();
  });
});
