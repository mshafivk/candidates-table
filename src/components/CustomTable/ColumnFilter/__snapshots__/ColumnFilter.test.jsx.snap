// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnFilter Component should render properly 1`] = `
<select
  onChange={[Function]}
>
  <option
    key="0"
    value="ALL"
  >
    ALL
  </option>
  <option
    key="1"
  />
</select>
`;
